### PR TITLE
Always wrap center coordinate to the range [-180, 180]

### PR DESCRIPTION
**Pull request by mpulkki-mapbox**
Panning the map doesn't automatically wrap the center point to the range [-180, 180]. User can move the map outside of the screen so that nothing gets rendered. This is possible as `center.wrap()` is invoked only during easing of the fling gesture and not at all if user pans the map without leaving any inertia.

https://user-images.githubusercontent.com/55925868/151778009-8ab34a71-1e00-410f-8d8f-24b1023f8f05.mp4

This PR fixes the issue by wrapping the center point in `transform._translateCameraConstrained()`. Additionally the `pointCoordinate` implementation for the globe view had to be updated to return non-wrapped coordinates to reflect the change.

Fixes the following issue https://github.com/mapbox/mapbox-gl-js/issues/11354 which was caused by mixed usage of wrapped and unwrapped coordinates in tile coverage computations.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes
 - [x] write tests for all new functionality
 - [x] manually test the debug page
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Fix map center not being wrapped properly after a drag gesture.</changelog>`

### CODE REVIEW

File: src/util/util.js [hunk 1]
```diff
@@ -239,6 +239,19 @@ export function wrap(n: number, min: number, max: number): number {
...
+ *
+ * @param {*} a First angle in degrees
+ * @param {*} b Second angle in degrees
+ * @returns Shortest angle
+ * @private
+ */
+export function shortestAngle(a: number, b: number): number {
+    const diff = (b - a + 180) % 360 - 180;
+    return diff < -180 ? diff + 360 : diff;
+}
```
Comment (SnailBones) [inline src/util/util.js:253]: Why is this function in `util.js` rather than inlined? Will it be used in other places?

Comment (mpulkki-mapbox) [reply src/util/util.js:253]: Finding signed distance between two angles is a quite common and useful operation and a good candidate for having its own utility function. I couldn't find any good reasons for inlining it as the code path is not particularly hot and it's easier to test too. Would you like to see it inlined instead?

Comment (SnailBones) [reply src/util/util.js:253]: Those all seem like good reasons. Great work @mpulkki-mapbox!

**Review by SnailBones — APPROVED** [review]

**Review by karimnaaji — APPROVED** [review]
LGTM! Could you also make sure to update the changelog entry for that change?

**Pull request merged**